multiview/cluster/mv_k_means.py

```diff
@@ -0,0 +1,262 @@
+            new_partitions = np.argmin(distances, axis=0).flatten()
+
+            # Recompute the objective function
+            o_funct = 0
+            for cl in range(self._k):
+                vecs = Xs[view][(partitions[view] == cl)]
+                dist = np.linalg.norm(vecs - self._centroids[view][cl], axis=1)
+                o_funct += np.sum(dist)
+
+            # Track of number of iterations without improvement
+            if(o_funct < objective[view]):
+                objective[view] = o_funct
+                iter_stall = 0
+            else:
+                iter_stall += 1
+
+        v1_consensus = list()
+        v2_consensus = list()
+
+        for clust in range(self._k):
+
+            v1_distances = self._compute_distance(Xs[0], self._centroids[0])
+            v1_partitions = np.argmin(v1_distances, axis=0).flatten()
+            v2_distances = self._compute_distance(Xs[1], self._centroids[1])
+            v2_partitions = np.argmin(v2_distances, axis=0).flatten()
+
+            part_indices = (v1_partitions == clust) * (v2_partitions == clust)
+
+            if (np.sum(part_indices) != 0):
+                cent1 = np.mean(Xs[0][part_indices], axis=0)
+                v1_consensus.append(cent1)
+
+                cent2 = np.mean(Xs[1][part_indices], axis=0)
+                v2_consensus.append(cent2)
+
+        self._centroids[0] = np.vstack(v1_consensus)
+        self._centroids[1] = np.vstack(v2_consensus)
+
+        return self
+
+    def predict(self, Xs):
+
+        '''
+        Predict the cluster labels for the data.
+
+        Parameters
+        ----------
+        Xs : list of array_likes
+            - Xs shape: (2,)
+            - Xs[0] shape: (n_samples, n_features_i)
+            This list must be of size 2, corresponding to the two
+            views of the data. The two views can each have a different
+            number of features, but they must have the same number of samples.
+
+        Returns
+        -------
+        predictions : array-like, shape (n_samples,)
+            The predicted cluster labels for each sample.
+
+        '''
+
+        Xs = check_Xs(Xs)
+
+        dist1 = self._compute_distance(Xs[0], self._centroids[0])
+        dist2 = self._compute_distance(Xs[1], self._centroids[1])
+        dist_metric = dist1 + dist2
+        predictions = np.argmin(dist_metric, axis=0).flatten()
+
+        return predictions
+
+    def fit_predict(self, Xs, patience=5, max_iter=None):
+
+        '''
+        Fit the cluster centroids to the data and then
+        predict the cluster labels for the data.
+
+        Parameters
+        ----------
+        Xs : list of array_likes
+            - Xs shape: (2,)
+            - Xs[0] shape: (n_samples, n_features_i)
+            This list must be of size 2, corresponding to the two views
+            of the data. The two views can each have a different number
+            of features, but they must have the same number of samples.
+
```

Comment:
add descriptions of other input arguments to parameters. It looks like they're in the Returns section?